metrics/statsd.go

```diff
@@ -47,6 +48,21 @@ func (m *StatsdMonitor) RecordRun(run agent.Run) {
 	)
 }
 
+// RecordSink records a individual sinks behavior in a run
+func (m *StatsdMonitor) RecordSink(recipeName, sourceName string, sink agent.PluginRun) {
```

Comment:
let's generalise this to `RecordPlugin`, and let's use parameters only.

We only need, `recipe name`, `plugin name`, `success`, `plugin type`.

We can skip source name because we already have recipe name, that should be enough.